### PR TITLE
Include more filetypes

### PR DESCRIPTION
**Pull request by versecafe**
Release Notes:

- Expanded filetype coverage for less common formats
- Coverage for common xcode/obj-c(++) development
- Coverage HTML templating filetypes across languages

### CODE REVIEW

**Comment by maxdeviant** [comment]
One question I have is whether all of these file extensions should be in the base Zed configuration.

How many people are actually using extensions like `php3`, `php4`, and `php5`?

It's possible to configure file associations via the settings, so I don't think it's critical that we have _every_ possible extension under the sun in the base config.

**Comment by versecafe** [comment]
> One question I have is whether all of these file extensions should be in the base Zed configuration.
> 
> How many people are actually using extensions like `php3`, `php4`, and `php5`?
> 
> It's possible to configure file associations via the settings, so I don't think it's critical that we have _every_ possible extension under the sun in the base config.

the numbered php files are oddly common in long running php projects, but I can remove them, also when going through I noticed that even currently specified patterns like `.eslint.js` and `.eslint.json` just end up as js or json, the current parser doesn't do closest match

**Comment by versecafe** [comment]
@maxdeviant what's your opinion on using the same icon for c++ and c# since the c++ icon is just a chip into the c one with the icon size the # is basically invisible so I've recycled c for obj c and c++ for obj-c++ and c#

File: assets/icons/file_icons/blender.svg
Comment (maxdeviant) [inline assets/icons/file_icons/blender.svg:1]: We aren't accepting new file icons, at this time.

Comment (versecafe) [reply assets/icons/file_icons/blender.svg:1]: pulled them

**Comment by versecafe** [comment]
@maxdeviant This should be good to go now and finally React Native projects and Obj-C have gone from most icons missing to full coverage, plus a bunch of other small improvements with file extensions that got missed

### Old
image

### New

image

**Comment by maxdeviant** [comment]
@versecafe Thanks for your work on this PR.

I think we want to be a bit more selective about what file extensions we add out of-the-box, and focus primarily on widely-used file extensions that only have a predominantly single file type associated with them

Since it is possible to add file icon associations in user/project settings, we don't need to have everything under the sun in the base config.

**Pull request closed without merging**